### PR TITLE
feat: add fees support for main net txs

### LINKED ISSUE #175: [FEATURE] Provide fee for transactions
Zboxcli sends transactions with 0 fees, it won't work on mainnet
E.g: `newallocation` has no way to provide fees and internally uses:
```
func smartContractTxnValue(sn transaction.SmartContractTxnData, value int64) (hash, out string, err error) {
	return smartContractTxnValueFee(sn, value, 0)
}
```

Proposal: to make optional key `--withFees = ` to provide fee and if not provided use `min_fee` as default.

See: 
- https://github.com/0chain/zwalletcli/issues/85
- https://0chain.slack.com/archives/G01FYNN6DK4/p1645097379703449

### PR DESCRIPTION
**Pull request by bisakhmondal**
A brief description of the changes in this PR:

Tx fee

Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/zboxcli/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

Associated PRs (Link as appropriate):
- 0chain: https://github.com/0chain/0chain/pull/1753
- blobber:
- gosdk: https://github.com/0chain/gosdk/pull/623
- system_test:
- zwalletcli: https://github.com/0chain/zwalletcli/pull/184
- Other: ...

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ go 1.18
 
 require (
 	github.com/0chain/errors v1.0.3
-	github.com/0chain/gosdk v1.8.10-0.20221026040749-2a049c31338c
+	github.com/0chain/gosdk v1.8.10-0.20221031223116-98b77c9a9267
```
Comment (bisakhmondal) [inline go.mod]: This uses gosdk commit 98b77c9a9267696f18b89230b87cb0b192ae2c9c
```
🔥➜ zboxcli git:(chain-fee) go get github.com/0chain/gosdk@98b77c9a9267696f18b89230b87cb0b192ae2c9c
```

**Review by peterlimg — APPROVED** [review]
lgtm

**Review by dabasov — APPROVED** [review]
lgtm

**Comment by peterlimg** [comment]
@bisakhmondal please fix the conflicts

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee-integration  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | ABILITY TO SKIP TESTS IS DISABLED. SKIP_TESTS OPTION IGNORED.  |
| system-tests  | chain-fee  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | ABILITY TO SKIP TESTS IS DISABLED. SKIP_TESTS OPTION IGNORED.  |
| system-tests  | chain-fee-remove-excess-faucet  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | ABILITY TO SKIP TESTS IS DISABLED. SKIP_TESTS OPTION IGNORED.  |
| system-tests  | chain-fee  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | chain-fee  |
| zwalletcli  | chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee-v2  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0dns  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chain-fee  |
| gosdk  | chain-fee  |
| zboxcli  | chain-fee  |
| zwalletcli  | feat/chain-fee  |
| 0chain  | chain-fee-v2  |
| blobber  | chain-fee  |
| authorizer  | chain-fee  |
| 0box  | chain-fee  |
| 0dns  | staging  |

**Pull request merged**